empyrical/stats.py

```diff
@@ -1005,6 +993,214 @@ def cagr(returns, period=DAILY, annualization=None):
+        - This is in the same style as returns.
+    period : str, optional
+        Defines the periodicity of the 'returns' data for purposes of
+        annualizing. Value ignored if `annualization` parameter is specified.
+        Defaults are:
+            'monthly':12
+            'weekly': 52
+            'daily': 252
+    Returns
+    -------
+    float, np.nan
+        The capture ratio.
+    """
+    return (annual_return(returns, period=period) /
+            annual_return(factor_returns, period=period))
+
+
+def up_capture(returns, factor_returns, **kwargs):
+    """
+    Compute the capture ratio for periods when the benchmark return is positive
+
+    Parameters
+    ----------
+    see documentation for `capture`.
+
+    Returns
+    -------
+    float, np.nan
+    """
+    return up(returns, factor_returns, functions=[capture], **kwargs)
+
+
+def down_capture(returns, factor_returns, **kwargs):
+    """
+    Compute the capture ratio for periods when the benchmark return is negative
+
+    Parameters
+    ----------
+    see documentation for `capture`.
+
+    Returns
+    -------
+    float, np.nan
+    """
+    return down(returns, factor_returns, functions=[capture], **kwargs)
+
+
+def up_down_capture(returns, factor_returns, **kwargs):
+    """
+    Computes the ratio of up_capture to down_capture.
+
+    Parameters
+    ----------
+    see documentation for `capture`.
+
+    Returns
+    -------
+    float
+        the updown capture ratio
+    """
+    return (up_capture(returns, factor_returns, **kwargs) /
+            down_capture(returns, factor_returns, **kwargs))
+
+
+def up_alpha_beta(returns, factor_returns, **kwargs):
+    """
+    Computes alpha and beta for periods when the benchmark return is positive.
+
+    Parameters
+    ----------
+    see documentation for `alpha_beta`.
+
+    Returns
+    -------
+    list
+        float
+            Alpha.
+        float
+            Beta.
+    """
+    return up(returns, factor_returns, functions=[alpha_beta], **kwargs)
+
+
+def down_alpha_beta(returns, factor_returns, **kwargs):
+    """
+    Computes alpha and beta for periods when the benchmark return is negative.
+
+    Parameters
+    ----------
+    see documentation for `alpha_beta`.
+
+    Returns
+    -------
+    list
+        float
+            Alpha.
+        float
+            Beta.
+    """
+    return down(returns, factor_returns, functions=[alpha_beta], **kwargs)
+
+
+def roll_up_capture(returns, factor_returns, window=10, **kwargs):
+    """
+    Computes the up capture measure over a rolling window.
+
+    Parameters
+    ----------
+    see documentation for `capture` (pass all args, kwargs required)
+
+    window : int, required
+        Size of the rolling window in terms of the periodicity of the data.
+        - eg window = 60, periodicity=DAILY, represents a rolling 60 day window
+    """
+    return roll(returns, factor_returns, window=window, functions=[
```

Comment:
I think your idea about a single function makes the most sense. The `functions=[]` was sort of over-engineered. I changed it all to take a `function` kwarg in 	71ea444